channel/src/main/java/com/synopsys/integration/alert/channel/jira2/common/delegate/JiraIssueCreator.java

```diff
@@ -0,0 +1,129 @@
+import com.synopsys.integration.jira.common.model.response.IssueResponseModel;
+import com.synopsys.integration.rest.exception.IntegrationRestException;
+
+public abstract class JiraIssueCreator<T> extends IssueTrackerIssueCreator<String> {
+    private final JiraErrorMessageUtility jiraErrorMessageUtility;
+    private final JiraIssueAlertPropertiesManager issuePropertiesManager;
+    private final String issueCreatorDescriptorKey;
+
+    protected JiraIssueCreator(
+        IssueTrackerChannelKey channelKey,
+        IssueTrackerIssueCommenter<String> commenter,
+        IssueTrackerCallbackInfoCreator callbackInfoCreator,
+        JiraErrorMessageUtility jiraErrorMessageUtility,
+        JiraIssueAlertPropertiesManager issuePropertiesManager,
+        String issueCreatorDescriptorKey
+    ) {
+        super(channelKey, commenter, callbackInfoCreator);
+        this.jiraErrorMessageUtility = jiraErrorMessageUtility;
+        this.issuePropertiesManager = issuePropertiesManager;
+        this.issueCreatorDescriptorKey = issueCreatorDescriptorKey;
+    }
+
+    @Override
+    protected final ExistingIssueDetails<String> createIssueAndExtractDetails(IssueCreationModel alertIssueCreationModel) throws AlertException {
+        T creationRequest = createIssueCreationRequest(alertIssueCreationModel);
+        try {
+            IssueCreationResponseModel issueCreationResponseModel = createIssue(creationRequest);
+            IssueResponseModel createdIssue = fetchIssue(issueCreationResponseModel.getKey());
+            IssueFieldsComponent createdIssueFields = createdIssue.getFields();
+
+            String issueUILink = JiraCallbackUtils.createUILink(createdIssue);
+            return new ExistingIssueDetails<>(createdIssue.getId(), createdIssue.getKey(), createdIssueFields.getSummary(), issueUILink);
+        } catch (IntegrationRestException restException) {
+            throw jiraErrorMessageUtility.improveRestException(restException, issueCreatorDescriptorKey, getReporter(creationRequest));
+        } catch (IntegrationException intException) {
+            throw new AlertException("Failed to create an issue in Jira.", intException);
+        }
+    }
+
+    @Override
+    protected final void assignAlertSearchProperties(ExistingIssueDetails<String> createdIssueDetails, ProjectIssueModel alertIssueSource) throws AlertException {
+        JiraIssueSearchProperties searchProperties = createSearchProperties(alertIssueSource);
+        issuePropertiesManager.assignIssueProperties(createdIssueDetails.getIssueKey(), searchProperties);
+    }
+
+    protected abstract T createIssueCreationRequest(IssueCreationModel alertIssueCreationModel) throws AlertException;
+
+    protected abstract IssueCreationResponseModel createIssue(T alertIssueCreationModel) throws IntegrationException;
+
+    protected abstract IssueResponseModel fetchIssue(String createdIssueKey) throws IntegrationException;
+
+    protected abstract String getReporter(T creationRequest);
```

Comment:
Can we call this `extractReporter`? I feel like `get` makes this seem like it could be performing a GET request.

Reply:
I made `fetchIssue` over the concern that `get` might not imply a rest request. So `getIssue` and `extractReporter`?

Reply:
I actually prefer `fetch`/`retrieve` instead of `get` for I/O (network, database, filesystem). I would like `get` to only be used for "getters". The only reason I mentioned a GET request in this case is because `getReporter` takes an argument that has the word "request" in it.

Sorry for making it overly complicated 😅 

To summarize:
- `getIssue` 👎🏻 
- `fetchIssue` 👍🏻 
- `getReporter` 👎🏻 
- `extractReporter` 👍🏻